app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,17 +11,30 @@ def self.call(*args)
       end
 
       def call
-        # This method should find the user based on the `receiver_id`.
-        # It should then determine the appropriate Broadcast for a user,
-        # based on the `created_at` and the different conditions for sending a notification.
-        # `welcome_broadcast = ...`
+        return if commented_on_welcome_thread? || received_notification?
 
-        # Once it has the appropriate Broadcast to be sent, it should send a notification for it:
-        # `Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)`
+        Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)
+      end
+
+      def received_notification?
+        Notification.exists?(notifiable: welcome_broadcast, user: user)
+      end
+
+      def commented_on_welcome_thread?
+        welcome_threads = Article.where("title LIKE 'Welcome Thread - %'").order(created_at: :desc).first
```

Comment:
I wish there was a better way to tag welcome threads, as this would pick up any user generated thread called "Welcome Thread - %". 

Not for this PR but I might have a proposal for the future:

- what if we create a tag `welcomethread` that can only be used by admins
- write a data update script to add this tag to all existing welcome thread articles
- change this to something that loads only articles with that tag

The reason is that `LIKE` on hundreds of thousands of articles might become slow (the `title` field is not even indexed :-/) 

ps. I think this was probably reverted as @vaidehijoshi suggestion to rename the variable is not applied anymore

Reply:
@rhymes these are great suggestions! I definitely have some ideas for how to improve this generator down the road, which I think will be easier to do (and will become more obvious which approach to take) as we start working on adding more broadcasts (which will mean different kinds of welcome notifications) to this generator class.

Love the idea of tagging the welcome threads. I’ll add it as a separate ticket to the team project board since it’s so out of scope for this PR, but something we do want to do!

Reply:
Thanks for the feedback! I really love the idea of tagging Welcome Threads, too. 😄

Reply:
What if we use the [same implementation](https://github.com/thepracticaldev/dev.to/blob/2b442ca8b69e58832cc82b9dfe9ad8db42523b9d/app/controllers/pages_controller.rb#L88) we used for out `/welcome`? It has exact logic we are looking for

Reply:
Ahhh! Nice find, @maestromac! 👏 I didn't see that there.

Seems like we could copy-pasta that into here and then rework it as a scope when we go about fixing #6702?

Reply:
Good eyes @maestromac ! 

..."copy-pasta" gets me every time. 😂